taqueria-vscode-extension/src/test/suite/extension.test.ts

```diff
@@ -5,102 +5,112 @@ import * as vscode from 'vscode';
-import { sleep } from './utils/utils';
 
-const projectRoot = path.resolve(__dirname, '../../../');
-const testProjectDestination = `${projectRoot}/out/vscode-taq-test-project`;
-const testProjectSource = `${projectRoot}/src/test/suite/data/vscode-taq-test-project`;
-
-describe('Extension Test Suite', () => {
+const homedir = require('os').homedir();
+const sourceFilesRoot = path.resolve(__dirname, '../../../');
+const testProjectRoot = path.resolve(homedir, 'TVsCE_e2e');
+const testProjectSource = `${sourceFilesRoot}/src/test/suite/data`;
+const ligoContractFileSource = `${testProjectSource}/hello-tacos.mligo`;
+
+const testProjectDestination = `${testProjectRoot}/vscode-taq-test-project`;
+const ligoContractFileDestination = `${testProjectDestination}/contracts/hello-tacos.mligo`;
+
+const originalMethods = {
+	'window.showInformationMessage': vscode.window.showInformationMessage,
+	'window.showQuickPick': vscode.window.showQuickPick,
+};
+
+let vscodeMock: typeof vscode;
+let mockedReturnValues = {
+	openDialogMockValues: [''],
+	openDialogMockIndex: 0,
+};
+
+function setOpenDialogMocks(...mockedPaths: string[]) {
+	mockedReturnValues.openDialogMockValues = mockedPaths;
+	mockedReturnValues.openDialogMockIndex = 0;
+}
+
+function getNextOpenDialogUri() {
+	if (
+		!mockedReturnValues.openDialogMockValues
+		|| mockedReturnValues.openDialogMockIndex >= mockedReturnValues.openDialogMockValues.length
+	) {
+		throw new Error('We have already used all mocked values.');
+	}
+	const mockedValue = mockedReturnValues.openDialogMockValues[mockedReturnValues.openDialogMockIndex];
+	mockedReturnValues.openDialogMockIndex++;
+	return mockedValue;
+}
+
+let choosePlugin: string = '@taqueria/plugin-smartpy';
+
+const mockedMethods = {
+	'window.showInformationMessage': (msg: string) => Promise.resolve(console.log(msg)),
+	'window.showQuickPick': () => Promise.resolve([choosePlugin]),
+	'window.showOpenDialog': () => {
+		return Promise.resolve([Uri.file(getNextOpenDialogUri())]);
+	},
+};
+vscodeMock = MockedObject.make(vscode, mockedMethods);
+
+describe('Extension Test Suite', async () => {
 	before(async () => {
 		const context: ExtensionContext = {
 			subscriptions: [],
 		} as any;
-		await taqueriaExtension.activate(context, {
-			vscode: MockedObject.make(vscode, {
-				'window.showInformationMessage': (msg: string) => Promise.resolve(console.log(msg)),
-			}),
-		});
-		await fse.copy(testProjectSource, testProjectDestination);
+
+		await taqueriaExtension.activate(context, { vscode: vscodeMock });
 
 		vscode.window.showInformationMessage('Start all tests.');
 	});
 
-	it('Verify that Taqueria Initiate will init new taquifed  project ', async () => {
+	it('Verify that Taqueria commands present in the command pallet list ', async () => {
 		await vscode.commands.getCommands(true).then(allCommands => {
 			const taqCommands = allCommands.filter(command => command.toLowerCase().includes('taq'));
 			assert.notEqual(taqCommands, undefined);
```

Comment:
If we are checking that the commands are there I think we can use this test as a list containing all those commands so we know if anything changes rather than just checking that the commands are not undefined

Reply:
I updated test logic to check list of commmands.